spec/controllers/books_controller_spec.rb

```diff
@@ -107,7 +107,16 @@
     let(:send_request){ put :update, params}
     it "updates the data in a Book object" do
       send_request
-      expect(assigns(:book).reload.title).to eq(valid_params[:title])
+      expect(assigns(:book).title).to eq(valid_params[:title])
+    end
+  end
+
+  describe "PUT #update" do
+    let(:params){ { id: book.id } }
+    let(:send_request){ get :show, params}
```

Comment:
You can change this to `let!(:send_request)` and then take it out of the `it` block. More info here: https://relishapp.com/rspec/rspec-core/v/2-6/docs/helper-methods/let-and-let